File: core/sample.py
"""Структуры сэмплов и сырых событий ввода."""

from __future__ import annotations

from dataclasses import dataclass, field


@dataclass
class GamepadSample:
    timestamp_ns: int
    axes: dict[str, float] = field(default_factory=dict)
    buttons: dict[int, bool] = field(default_factory=dict)
    hat: tuple[int, int] = (0, 0)


@dataclass
class RawInputEvent:
    timestamp_ns: int
    event_type: str
    code: str
    value: int | float
    test_id: str = ""


@dataclass
class DeviceInfo:
    name: str = ""
    path: str = ""
    vendor_id: int | None = None
    product_id: int | None = None
    axis_profile: str = ""
    connected: bool = False

File: src-tauri/src/sample.rs
use std::collections::HashMap;

#[derive(Clone, Debug)]
pub struct GamepadSample {
    pub timestamp_ns: u64,
    pub axes: HashMap<String, f64>,
    #[allow(dead_code)]
    pub buttons: HashMap<u8, bool>,
    #[allow(dead_code)]
    pub hat: (i32, i32),
}

#[derive(Clone, Debug)]
pub struct RawInputEvent {
    pub timestamp_ns: u64,
    pub event_type: String,
    pub code: String,
    pub value: String,
    pub test_id: String,
}
